datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -460,12 +460,13 @@ impl FileFormat for ParquetFormat {
             metadata_size_hint = Some(metadata);
         }
 
-        let table_schema = TableSchema::new(
-            Arc::clone(conf.file_schema()),
-            conf.table_partition_cols().clone(),
-        );
-        let mut source = ParquetSource::new(table_schema)
-            .with_table_parquet_options(self.options.clone());
+        let mut source = conf
+            .file_source()
+            .as_any()
+            .downcast_ref::<ParquetSource>()
+            .cloned()
+            .expect("should be a parquet source");
```

Comment:
it seems like this would be better if it were an internal error rather than panic 

The idea of using the existing source rather than creating a new one makes a lot of sense though